### ISSUE TITLE
Unable to exclude xunit assemblies from TeamCity dotcover results

### ISSUE THREAD

**Issue by WattsC-90** [xunit/xunit#1748]
Hi,
So I apologise if this sounds, or turns out not to be a bug on xunit but bear with me and it may become evident why I have raised this ticket..

Our TeamCity builds with 2.2.0 use dotcover for code coverage, xunit can be excluded without issue there, so it doesn't skew the results incorrectly. If I upgrade the xunit nuget to 2.3.1, without changing anything in the build step, the xunit assemblies will now be included in the coverage results. The reverse happens on downgrade back to 2.2.0, the coverage report now excludes the xunit assemblies.

Like I say, to me this seems like an xunit issue, based on the xunit version being the only bit that changed, but I am more than happy to be wrong!

I have to try and keep company specific information out of here, but if you require/have more information/questions, I will try my best to put it up here.

Thanks, Chris

**Comment by bradwilson** [comment]
Without any experience with dotcover and/or its integration with TeamCity, it's hard to know where this exclusionary behavior comes from (and whether it's something we're doing, as opposed to 3rd party code).

Can you provide some documentation (and ideally a repro project) that shows how everything is wired up into TeamCity?

**Comment by WattsC-90** [comment]
Thats going to be a bit challenging as its corporate/business builds.. but I am using a public metarunner in teamcity. See here: https://github.com/JetBrains/meta-runner-power-pack/tree/master/xunit.net

The C# itself is built using vs2015, if that helps (probably not..) I can see if i can find some time to create a complete dummy project that causes this issue in the next week or so

**Comment by bradwilson** [comment]
Thanks, I'll see if this helps me understand what's going on and whether I can get a simple repro. What versions of TeamCity and dotcover are you running, in case that matters?

**Comment by WattsC-90** [comment]
dotcover version is 2017.1.20170407.131846
i believe it ships by default with the TeamCity version we are using: 2017.1.1 (build 46654)
long overdue an upgrade!

**Comment by onovotny** [comment]
I have a feeling dotCover includes any type it has PDB's for.

With Rx.NET, I had to add an exclude for xUnit like this:

If you're using it for WinForms/WPF, you can set https://github.com/dotnet/reactive/blob/rel/v4.0/Rx.NET/Source/build-new.ps1#L89

```powershell

$dotnet = "$env:ProgramFiles\dotnet\dotnet.exe"
.\packages\JetBrains.dotCover.CommandLineTools\tools\dotCover.exe analyse /targetexecutable="$dotnet" /targetworkingdir="$testDirectory" /targetarguments="test -c $configuration --no-build --no-restore --filter `"SkipCI!=true`"" /Filters="+:module=System.Reactive;+:module=Microsoft.Reactive.Testing;+:module=System.Reactive.Observable.Aliases;-:type=Xunit*" /DisableDefaultFilters /ReturnTargetExitCode /AttributeFilters="System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute" /Output="$outputFile" /ReportType=DetailedXML /HideAutoProperties
```

**Comment by WattsC-90** [comment]
Hi @onovotny , I added an exclusion for XUnit* to the build step, which i think is what you were saying in your code excerpt? I am building a windows service, no UI components. Like I said in the original ticket, when I change the version back to 2.2.0, with the exclude assemblies filter set to XUnit*, it will exclude all XUnit dlls, at >2.3.x it will include them, regardless of that filter. It is very possible that it is something to do with TeamCity/the metarunner I am using, but it seemed logical to ask here first as I was able to simply change the version of XUnit and get them to appear/disappear
